mindsdb/integrations/handlers/openbb_handler/openbb_handler.py

```diff
@@ -0,0 +1,164 @@
+
+    def connect(self) -> bool:
+        """Connects with OpenBB account through personal access token (PAT).
+
+        Returns none.
+        """
+        self.is_connected = False
+        obb.account.login(pat=self.PAT)
+
+        # Check if PAT utilized is valid
+        if obb.user.profile.active:
+            self.is_connected = True
+            return True
+
+        return False
+
+    def check_connection(self) -> StatusResponse:
+        """Checks connection to OpenBB accounting by checking the validity of the PAT.
+
+        Returns StatusResponse indicating whether or not the handler is connected.
+        """
+
+        response = StatusResponse(False)
+
+        try:
+            if self.connect():
+                response.success = True
+
+        except Exception as e:
+            log.logger.error(f"Error connecting to OpenBB Platform: {e}!")
+            response.error_message = e
+
+        self.is_connected = response.success
+        return response
+
+    def _process_cols_names(self, cols: list) -> list:
+        new_cols = []
+        for element in cols:
+            # If the element is a tuple, we want to merge the elements together
+            if isinstance(element, tuple):
+                # If there's more than one element we want to merge them together
+                if len(element) > 1:
+                    # Prevents the case where there's a multi column index and the index is a date
+                    # in that instance we will have ('date', '') and this avoids having a column named 'date_'
+                    new_element = "_".join(map(str, element)).rstrip("_")
+                    new_cols.append(new_element)
+                else:
+                    new_cols.append(element[0])
+            else:
+                new_cols.append(element)
+        return new_cols
+
+    def _openbb_fetcher(self, params: Dict = None) -> pd.DataFrame:
+        """Gets aggregate trade data for a symbol based on given parameters
+
+        Returns results as a pandas DataFrame.
+
+        Args:
+            params (Dict): Trade data params (symbol, interval, limit, start_time, end_time)
+        """
+        self.connect()
+
+        try:
+            if params is None:
+                log.logger.error("At least cmd needs to be added!")
+                return pd.DataFrame()
+
+            # Get the OpenBB command to get the data from
+            cmd = params.pop("cmd")
+
+            args = ""
+            # If there are parameters create arguments as a string
+            if params:
+                for arg, val in params.items():
+                    args += f"{arg}={val},"
+
+                # Remove the additional ',' added at the end
+                if args:
+                    args = args[:-1]
+
+            # Recreate the OpenBB command with the arguments
+            openbb_cmd = f"{cmd}({args})"
+
+            # Execute the OpenBB command and return the OBBject
+            OBBject = eval(openbb_cmd)
```

Comment:
Is there some other function in the openbb that we can use to execute the code instead of using eval?

Reply:
Also, can we use another name as openbb_object to be consistent with the current python naming convention?

Reply:
Yes regarding openbb_object.

No regarding eval. We can maybe ensure that the user passes a valid func before running anything? Thoughts?